### PR TITLE
fix(core): Amplify.configure dispatches Hub event with unparsed config object

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of how you validated changes

* unit testing

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/singleton/Amplify.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export class AmplifyClass {
 			'core',
 			{
 				event: 'configure',
-				data: resourcesConfig,
+				data: this.resourcesConfig,
```
Comment (jimblanc) [inline packages/core/src/singleton/Amplify.ts:75]: Is this technically a breaking change?

Comment (HuiSF) [reply packages/core/src/singleton/Amplify.ts:75]: Not really, this change is for fixing when call `.configure()` on the Amplify singleton imported directly from core (internal usage).

When calling `.configure` on the Amplify singleton proxy imported from `aws-amplify`, the resourceConfig param that reaches the core singleton, is always parsed.

Comment (jimblanc) [reply packages/core/src/singleton/Amplify.ts:75]: What if customers are using the data from the `configure` event though?

Comment (HuiSF) [reply packages/core/src/singleton/Amplify.ts:75]: > What if customers are using the data from the `configure` event though?

If they are using `Amplify` imported from `aws-amplify` (which should be the case as we don't document to use `Amplify` imported from core), the `configure` event gets the data as `ResourceConfig`.

Comment (jimblanc) [reply packages/core/src/singleton/Amplify.ts:75]: Got it, thanks for confirming. Just wanted to check!

**Review by jimblanc — APPROVED** [review]

**Review by kvramyasri7 — APPROVED** [review]

**Pull request merged**